### PR TITLE
Add support for `monitor_metrics`

### PR DESCRIPTION
**Pull request by zioproto**
Add support for `monitor_metrics`

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster#monitor_metrics

## Issue number

#340 

## Checklist before requesting a review
- [X] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [X] I have executed pre-commit on my machine
- [X] I have passed pr-check on my machine

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -623,6 +623,15 @@ variable "microsoft_defender_enabled" {
   nullable    = false
 }
 
+variable "monitor_metrics" {
+  type = object({
+    annotations_allowed = optional(string)
+    labels_allowed      = optional(string)
+  })
+  description = "(Optional) Specifies a Prometheus add-on profile for the Kubernetes Cluster"
```
Comment (lonegunmanb) [inline variables.tf:631]: Could we enrich the description like this:?

```hcl
description = <<-EOT
(Optional) Specifies a Prometheus add-on profile for the Kubernetes Cluster
object({
    annotations_allowed = "(Optional) Specifies a comma-separated list of Kubernetes annotation keys that will be used in the resource's labels metric."
    labels_allowed      = "(Optional) Specifies a comma-separated list of Kubernetes annotation keys that will be used in the resource's labels metric."
})
```

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @zioproto for opening this pr! One comment.

File: variables.tf (hunk 1 above)
Comment (zioproto) [reply variables.tf]: done !

File: variables.tf [hunk 2]
```diff
@@ -623,6 +623,20 @@ variable "microsoft_defender_enabled" {
...
+  type = object({
+    annotations_allowed = optional(string)
+    labels_allowed      = optional(string)
+  })
+  description = <<-EOT
+  (Optional) Specifies a Prometheus add-on profile for the Kubernetes Cluster
+  object({
+    annotations_allowed = "(Optional) Specifies a comma-separated list of Kubernetes annotation keys that will be used in the resource's labels metric."
+    labels_allowed      = "(Optional) Specifies a Comma-separated list of additional Kubernetes label keys that will be used in the resource's labels metric."
+  })
```
Comment (lonegunmanb) [inline variables.tf:636]: We need an `EOT` here to end heredoc.

**Review by lonegunmanb — CHANGES_REQUESTED** [review]

**Pull request merged**